peer_client.go

```diff
@@ -333,10 +334,22 @@ func (c *PeerClient) getPeerRateLimitsBatch(ctx context.Context, r *RateLimitReq
 		err := &PeerErr{err: errors.New("already disconnecting")}
 		return nil, c.setLastErr(err)
 	}
+
+	// Wait for a response or context cancel
+	ctx2 := tracing.StartNamedScopeDebug(ctx, "Wait for response")
+	defer tracing.EndScope(ctx2, nil)
+
+	// Propagate the trace context along with the batched rate limit so
+	// peers can continue to report traces for this rate limit.
+	prop := propagation.TraceContext{}
```

Comment:
Make `prop` a global?

Reply:
Please avoid pre-optimization at all costs. Golang is very good at optimizing these types of things.

Reply:
Noted.